src/yunohost/app.py

```diff
@@ -100,14 +101,13 @@ def app_fetchlist(url=None, name=None):
             raise MoulinetteError(errno.EINVAL,
                                   m18n.n('custom_appslist_name_required'))
 
-    list_file = '%s/%s.json' % (repo_path, name)
-    if os.system('wget "%s" -O "%s.tmp"' % (url, list_file)) != 0:
-        os.remove('%s.tmp' % list_file)
+    try:
+        urlretrieve(url, '%s/%s.json' % (repo_path, name))
+    except Exception as e:
+        # I don't know how to put e into the MoulinetteError stuff
+        print e
         raise MoulinetteError(errno.EBADR, m18n.n('appslist_retrieve_error'))
 
-    # Rename fetched temp list
-    os.rename('%s.tmp' % list_file, list_file)
```

Comment:
Yes because I'm not creating any .tmp file since I'm using urlretreive directly.